cvat/apps/engine/views.py

```diff
@@ -616,10 +614,38 @@ def data(self, request, pk):
                     status=status.HTTP_400_BAD_REQUEST)
             serializer = DataSerializer(data=request.data)
             serializer.is_valid(raise_exception=True)
+            data = {k: v for k, v in serializer.validated_data.items()}
+
+            tmp_path = os.path.join(settings.TMP_DATA_ROOT, pk)
```

Comment:
@klakhov , what is the reason to upload into a temporary directory? Why don't upload directly to upload_dir?

Reply:
The idea behind using tmp directory is that we are uploading files before creating task in database. So, if task is not created, we dont have access to upload_dir.
It is possible to change this logic to: create task, upload data to upload_dir, update data in database. Would it be the best choice?